### PR TITLE
converted batch select to es6

### PR DESCRIPTION
**Pull request by cam156**
Did this to test my knowledge after running through the es6 tutorial
Additionally added a jasmine test to make sure this feature does not break in the future

Changes proposed in this pull request:
* convert batch_select.js to batch_select.es6
* Include a jasmine test for batch_select.js

@samvera/hyrax-code-reviewers

### CODE REVIEW

File: app/assets/javascripts/hyrax/app.js [hunk 1]
```diff
@@ -134,5 +135,12 @@ Hyrax = {
     statistics: {},
 
     // initialized in hyrax/config.js
-    config: {}
+    config: {},
+
+    // Per Page select will submit its form to change records shown
+    batchSubmit: function () {
+        var BatchSubmit = require('hyrax/batch_select');
```
Comment (cam156) [inline app/assets/javascripts/hyrax/app.js]: @jcoyne This matches the other requires in the system.  Any idea why hound is yelling for this one? Should it just be ignored?

Comment (jcoyne) [inline app/assets/javascripts/hyrax/app.js]: Yeah, we're just ignoring these. I guess eventually we could convert app.js to app.es6 and do away with these altogether.

File: app/assets/javascripts/hyrax/app.js [hunk 2]
```diff
@@ -134,5 +135,12 @@ Hyrax = {
     statistics: {},
 
     // initialized in hyrax/config.js
-    config: {}
+    config: {},
+
+    // Per Page select will submit its form to change records shown
```
Comment (jcoyne) [inline app/assets/javascripts/hyrax/app.js]: I'm not sure I understand "Per Page select".  Is there another way to say this?

File: app/assets/javascripts/hyrax/app.js (hunk 1 above)
Comment (jcoyne) [inline app/assets/javascripts/hyrax/app.js]: Why are we calling the file `batch_select` but the constant `BatchSubmit`? Can we decide on submit or select and stick with it?

File: spec/javascripts/batch_select_spec.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+var BatchSubmit = require('hyrax/batch_select');
+
+describe("BatchSelect", function() {
```
Comment (jcoyne) [inline spec/javascripts/batch_select_spec.js]: 👏 for adding a test.

**Comment by jcoyne** [comment]
Looks mostly good. Just a few questions around clarity.

**Comment by mjgiarlo** [comment]
Looks like this is close but could use a couple small tweaks from @cam156

**Comment by cam156** [comment]
@jcoyne I updated the comment and decided to go with batch_select as that was the original file name.

**Pull request merged**